frontend/public/components/resource-pages.ts

```diff
@@ -319,13 +318,14 @@ export const baseDetailsPages = ImmutableMap<ResourceMapKey, ResourceMapValue>()
     ),
   );
 
-export const resourceDetailsPages = ImmutableMap<ResourceMapKey, ResourceMapValue>()
-  .merge(baseDetailsPages)
-  .withMutations((map) => {
-    plugins.registry.getResourceDetailsPages().forEach((page) => {
-      addResourcePage(map, page);
+export const getResourceDetailsPages = (pluginPages: ResourceDetailsPage[]) =>
```

Comment:
For those callers that don't need to work with plugin contributed pages:

```suggestion
export const getResourceDetailsPages = (pluginPages: ResourceDetailsPage[] = []) =>
```